### PR TITLE
React_Redux/  

### PR DESCRIPTION
**Pull request by paulshaun-kdtk**
# CHANGES

- installed react/redux library
- Refractored bookAddition.js to use reducers
- modified folder structure

### CODE REVIEW

File: src/redux/books/Books.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+                {' '}
+                {author}
+                {' '}
+              </p>
+            </span>
+
+            <ul className="list">
+              <li onClick={onRemoveBook} className="firstListItem">remove</li>
+              <li>edit</li>
+              <li>delete</li>
```
Comment (V-Blaze) [inline src/redux/books/Books.js]: - You have done great so far :+1: But there are requirements you need to meet before moving to the next milestone. :small_airplane: 

> Define a slice of state for books that:
Includes an array of books (initial state: empty array)
Includes a reducer that adds a book
Includes a reducer that removes a book

- [ ] The books.js file inside the redux folder is meant to contain the bookslice. that is the `Action types, action creators, and reducers for adding and removing a book`. This is meant to be a redux file and not a react component. the current content of your store.js is what you are meant to have in this file. I'll suggest you diligently go through the project requirements and implement them correctly. :+1: :100:

File: src/redux/categories/Components.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          !Welcome to the components page:
+          <br />
+          Please Note this part of the website is still under construction.
+        </h1>
+      </div>
+    </div>
+  );
+}
+
+export default Components;
```
Comment (V-Blaze) [inline src/redux/categories/Components.js]: > Define a slice of state for categories that:
Includes an array of categories (initial state: empty array)
Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)

- [ ] Just as I pointed out above about the book slice, this file is meant to contain the reducer and action for categories, it is not a react component but a redux file. Looking at the requirements above, you can see the guidelines and instructions on what you were meant to do in this file. I'll greatly suggest you go through the requirements diligently. :+1: :smile:

File: src/redux/store.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        books: state.books.map((book, i) => (i === action.payload.index
+          ? { ...book, progress: Math.min(book.progress + 5, 100) }
+          : book)),
+      };
+    default:
+      return state;
+  }
+}
+
+const store = createStore(booksReducer);
```
Comment (V-Blaze) [inline src/redux/store.js]: - [ ] In the store.js file, You were meant to use `configureStore` to create a reducer for `book` and `categories`.  You can always reference the lesson before this project or the official [REACT-REDUX](https://redux-toolkit.js.org/) documentation to get more broad knowledge. But I'll suggest you follow the project requirement and implement the redux store here. :small_airplane: :100:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @paulshaun-kdtk 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

giphy


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Installed `react-redux` and `redux-toolkit`✔
-  Correct Git flow👋 
-  Readme looks professonal✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        books: state.books.map((book, i) => (i === action.payload.index
+          ? { ...book, progress: Math.min(book.progress + 5, 100) }
+          : book)),
+      };
+    default:
+      return state;
+  }
+}
+
+export default booksReducer;
```
Comment (julie-ify) [inline src/redux/books/booksSlice.js]: - [ ] As the previous reviewer mentioned, the reducers should be set up with `createSlice` from redux toolkit. You can see [this tutorial](https://www.softkraft.co/how-to-setup-slices-with-redux-toolkit/) for create slice and how to export the reducers.

File: src/redux/categories/categoriesSlice.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+function categoriesReducer(state = initialState, action) {
+  switch (action.type) {
+    case 'CATEGORY':
+      return 'Under Construction';
+    default:
+      return state;
+  }
+}
+
+export default categoriesReducer;
```
Comment (julie-ify) [inline src/redux/categories/categoriesSlice.js]: - [ ] Kindly refactor your categories reducer to use slice as stated in the requirements.

**Review by julie-ify — CHANGES_REQUESTED** [review]
Hi @paulshaun-kdtk,

Good job so far!

## Highlights!
- Good code organization and store config ✔️
- Descriptive README ✔️ 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by topeogunleye — APPROVED** [review]
### Hi @paulshaun-kdtk, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
-  Set up reducers with createSlice from redux toolkit. ✔️ 
 
## [Optional] suggestions:
- For future pull requests, please notice that your branch name `reduxIntro` does not follow the convention of being named after a feature or action. For example, you could name your branch `addedReduxIntro`. This makes it easier to track the changes and the purpose of each branch. 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**